vertical-pod-autoscaler/enhancements/7571-support-pod-level-resources/README.md

```diff
@@ -0,0 +1,253 @@
+```
+
+It is also possible to combine both pod-level and container-level specifications. In this case, one container can define its own resource constraints - the `ide` container, while other containers (`tool1` and `tool2`) can dynamically use any remaining resources within the Pod's overall limit:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: workload
+  namespace: default
+spec:
+  resources:
+    limits:
+      memory: "1024Mi"
+      cpu: "4"
+  initContainers:
+    - image: tool1:latest
+      name: tool1
+      restartPolicy: Always
+    - image: tool2:latest
+      name: tool2
+      restartPolicy: Always
+  containers:
+    - name: ide
+      image: theia:latest
+      resources:
+        requests:
+          memory: "128Mi"
+          cpu: "0.5"
+        limits:
+          memory: "256Mi"
+          cpu: "1"
+```
+
+The benefits and implementation details of pod-level `resources` are described in [KEP-2837](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2837-pod-level-resource-spec/README.md). A related article is also available in the [Kubernetes documentation](https://kubernetes.io/docs/tasks/configure-pod-container/assign-pod-level-resources/).
+
+Currently, before this KEP, VPA computes recommendations only at the container level, and those recommendations are applied exclusively at the container level. With the new pod-level resources specifications, VPA should be able to read from the pod-level `resources` stanza, calculate pod-level recommendations, and scale at the pod level when users define pod-level `resources`.
+
+To address this, this KEP proposes extending the VPA object's `spec` and `status` fields, and introducing two new pod-level flags to set constraints directly at the pod level. For more details check the [Proposal](#proposal) section.
+
+### Goals
+
+* Add support for the pod-level resources stanza in VPA:
+  * Read pod-level values
+  * Calculate pod-level recommendations
+  * Apply recommendations at the pod level
+* Thoroughly document the new feature, focusing on areas that change default behaviors, in the [VPA documentation](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler/docs).
+
+## Design Details
+
+### Design Principles
+
+This section describes how VPA reacts based on where resources are defined (pod level, container level or both).
+
+Before this KEP, the recommender computes recommendations only at the container level, and VPA applies changes only to container-level fields. With this proposal, the recommender also computes pod-level recommendations in addition to container-level ones. Pod-level recommendations are derived from per-container usage and recommendations, typically by aggregating container recommendations. Container-level policy still influences pod-level output: setting `mode: Off` in `spec.resourcePolicy.containerPolicies` excludes a container from recommendations, and `minAllowed`/`maxAllowed` bounds continue to apply.
```

Comment:
Just want to sanity check this a little.

> typically by aggregating container recommendations

From what I can tell, the metric that metric-server provides is per-container.

So the idea is to leave the recommender as is, making per-container recommendations based on its per-container metric, and let the updater/admission-controller user an aggregated value for the Pod resources.

Is my understanding here right?